app/models/course/condition/assessment.rb

```diff
@@ -89,4 +94,13 @@ def required_assessments_for(conditional)
       ON ids.assessment_id = course_assessments.id
     SQL
   end
+
+  # Trigger for resolving the conditional for a course user
+  Course::Assessment::Submission.after_save do |submission|
```

Comment:
the only line I'm willing to accept within a block is to call a method. Do not put logic inside a block because blocks cannot be overridden.